lib/ApplicationFeatures/ApplicationServer.cpp

```diff
@@ -733,16 +779,24 @@ void ApplicationServer::unprepare() {
           << "caught unknown exception during unprepare of feature '"
           << feature->name() << "'";
     }
-    feature->state(FeatureState::UNPREPARED);
+    feature->state(ApplicationFeature::State::UNPREPARED);
     reportFeatureProgress(_state.load(std::memory_order_relaxed), feature->name());
   }
 }
 
 void ApplicationServer::wait() {
   LOG_TOPIC("f86df", TRACE, Logger::STARTUP) << "ApplicationServer::wait";
 
-  while (!_stopping) {
+  // wait here until beginShutdown has been called and finished
+  while (true) {
+    // wait until somebody calls beginShutdown and it finishes
     CONDITION_LOCKER(guard, _shutdownCondition);
+    
+    if (_abortWaiting) {
+      // yippieh!
+      break;
+    }
+
     guard.wait(100000);
```

Comment:
Using the chrono literals would make this slightly more readable and typesafe:
```suggestion
    using namespace std::chrono_literals;
    guard.wait(100000us);
```
or
```suggestion
    using namespace std::chrono_literals;
    guard.wait(100ms);
```